osu.Game/Overlays/Chat/ChatOverlayDrawableChannel.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Online.Chat;
+
+namespace osu.Game.Overlays.Chat
+{
+    public class ChatOverlayDrawableChannel : DrawableChannel
+    {
+        public ChatOverlayDrawableChannel(Channel channel)
+            : base(channel)
+        {
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            ChatLineFlow.Padding = new MarginPadding(0);
+        }
+
+        protected override Drawable CreateDaySeparator(DateTimeOffset time) => new ChatOverlayDaySeparator(time);
+
+        private class ChatOverlayDaySeparator : Container
+        {
+            private readonly DateTimeOffset time;
+
+            public ChatOverlayDaySeparator(DateTimeOffset time)
+            {
+                this.time = time;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OverlayColourProvider colourProvider)
+            {
+                RelativeSizeAxes = Axes.X;
+                AutoSizeAxes = Axes.Y;
+                Padding = new MarginPadding { Horizontal = 15, Vertical = 20 };
+                Child = new GridContainer
+                {
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    ColumnDimensions = new[]
+                    {
+                        new Dimension(GridSizeMode.Absolute, 200),
+                        new Dimension(GridSizeMode.Absolute, 15),
+                        new Dimension(),
+                    },
+                    Content = new[]
+                    {
+                        new[]
+                        {
+                            new GridContainer
```

Comment:
This nested grid feels awkward to me but it seemed like the easiest way to implement the design?
If there's a better way please let me know.